Validation/CTPPS/plugins/CTPPSDirectProtonSimulation.cc

```diff
@@ -539,9 +539,9 @@ void CTPPSDirectProtonSimulation::processProton(
 
         const auto *dg = geometry.sensor(detIdInt);
 
-        const auto x_half_width = dg->params().at(0);
-        const auto y_half_width = dg->params().at(1);
-        const auto z_half_width = dg->params().at(2);
+        const auto x_half_width = dg->getDiamondDimensions().xHalfWidth;
```

Comment:
same here as https://github.com/cms-sw/cmssw/pull/31240/files#r479139597 (speed is less of an issue here)